### ISSUE TITLE
FPTest.GetCachedPower error

### ISSUE THREAD

**Issue by nenin-sc** [fmtlib/fmt#1911]
Hello,
I compiled fmt 7.03 with mingw_w64 v8.0 gcc 10.2 and found  error in  "format-impl-test.exe"  (see attach). 

[fmterr.zip](https://github.com/fmtlib/fmt/files/5316161/fmterr.zip)

**Comment by vitaut** [comment]
Please post the error directly instead of attaching a .zip archive.

**Comment by nenin-sc** [comment]
It did not fit.

**gcc:**

```
gcc -v 
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=c:/mingw32/winlibs10.2/bin/../libexec/gcc/i686-w64-mingw32/10.2.0/lto-wrapper.exe
Target: i686-w64-mingw32
Configured with: ../configure --prefix=/R/winlibs32_stage/inst_gcc-10.2.0/share/gcc --build=i686-w64-mingw32 --host=i686-w64-mingw32 --with-pkgversion='MinGW-W64 i686-posix-dwarf, built by Brecht Sanders' --with-tune=generic --enable-checking=release --enable-threads=posix --with-dwarf2 --disable-sjlj-exceptions --disable-libunwind-exceptions --enable-serial-configure --disable-bootstrap --enable-host-shared --enable-plugin --enable-default-ssp --disable-rpath --disable-libstdcxx-pch --enable-libstdcxx-time=yes --disable-libstdcxx-debug --disable-version-specific-runtime-libs --with-stabs --disable-symvers --enable-languages=c,c++,fortran,lto,objc,obj-c++,d --disable-gold --disable-nls --disable-stage1-checking --disable-win32-registry --disable-multilib --enable-ld --enable-libquadmath --enable-libada --enable-libssp --enable-libstdcxx --enable-lto --enable-fully-dynamic-string --enable-libgomp --enable-graphite --enable-mingw-wildcard --with-mpc=/d/Prog/winlibs32_stage/custombuilt --with-mpfr=/d/Prog/winlibs32_stage/custombuilt --with-gmp=/d/Prog/winlibs32_stage/custombuilt --with-isl=/d/Prog/winlibs32_stage/custombuilt --enable-install-libiberty --enable-__cxa_atexit --without-included-gettext --with-diagnostics-color=auto --with-libiconv --with-system-zlib --with-build-sysroot=/R/winlibs32_stage/gcc-10.2.0/build_mingw/mingw-w64 --enable-large-address-aware
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 10.2.0 (MinGW-W64 i686-posix-dwarf, built by Brecht Sanders)
```

**Error:**

```
[ RUN      ] FPTest.GetCachedPower
D:\mingw\mklib10.2\fmt-7.0.3\test\format-impl-test.cc:269: Failure 
Value of: ldexp(static_cast<double>(fp.f), fp.e) 
  Actual: 1e-284 
Expected: pow(10, dec_exp) 
Which is: 1.0000000000000306e-284 
``` 
-------------------------------

Here was 4k+ lines of the error messages, including  

------------------------------

```
D:\mingw\mklib10.2\fmt-7.0.3\test\format-impl-test.cc:269: Failure
Value of: ldexp(static_cast<double>(fp.f), fp.e)
  Actual: 9.9999999999999998e-13
Expected: pow(10, dec_exp)
Which is: 1.0000000000000014e-12
D:\mingw\mklib10.2\fmt-7.0.3\test\format-impl-test.cc:269: Failure
Value of: ldexp(static_cast<double>(fp.f), fp.e)
  Actual: 9.9999999999999998e-13
Expected: pow(10, dec_exp)
Which is: 1.0000000000000014e-12
D:\mingw\mklib10.2\fmt-7.0.3\test\format-impl-test.cc:269: Failure
Value of: ldexp(static_cast<double>(fp.f), fp.e)
  Actual: 9.9999999999999998e-13
Expected: pow(10, dec_exp)
Which is: 1.0000000000000014e-12



`D:\mingw\mklib10.2\fmt-7.0.3\test\format-impl-test.cc:269: Failure
Value of: ldexp(static_cast<double>(fp.f), fp.e)
  Actual: 1.0000000000000001e+236
Expected: pow(10, dec_exp)
Which is: 1.0000000000000006e+236
[  FAILED  ] FPTest.GetCachedPower (54 ms)
[ RUN      ] FPTest.GetRoundDirection
[       OK ] FPTest.GetRoundDirection (0 ms)
[ RUN      ] FPTest.FixedHandler
[       OK ] FPTest.FixedHandler (0 ms)
[ RUN      ] FPTest.GrisuFormatCompilesWithNonIEEEDouble
[       OK ] FPTest.GrisuFormatCompilesWithNonIEEEDouble (0 ms)
[----------] 8 tests from FPTest (54 ms total)

[----------] 5 tests from FormatTest
[ RUN      ] FormatTest.ArgConverter
[       OK ] FormatTest.ArgConverter (0 ms)
[ RUN      ] FormatTest.FormatNegativeNaN
[       OK ] FormatTest.FormatNegativeNaN (0 ms)
[ RUN      ] FormatTest.StrError
[       OK ] FormatTest.StrError (0 ms)
[ RUN      ] FormatTest.FormatErrorCode
[       OK ] FormatTest.FormatErrorCode (0 ms)
[ RUN      ] FormatTest.CountCodePoints
[       OK ] FormatTest.CountCodePoints (0 ms)
[----------] 5 tests from FormatTest (0 ms total)

[----------] 2 tests from UtilTest
[ RUN      ] UtilTest.CountDigits
[       OK ] UtilTest.CountDigits (0 ms)
[ RUN      ] UtilTest.WriteFallbackUIntPtr
[       OK ] UtilTest.WriteFallbackUIntPtr (0 ms)
[----------] 2 tests from UtilTest (0 ms total)

[----------] Global test environment tear-down
[==========] 26 tests from 4 test cases ran. (58 ms total)
[  PASSED  ] 25 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] FPTest.GetCachedPower

 1 FAILED TEST 
```

**Comment by vitaut** [comment]
This is a test issue, you can ignore it for now.

**Comment by vitaut** [comment]
Fixed the test in #1913. Thanks for reporting.
